tests/python/gaia-ui-tests/gaiatest/tests/functional/browser/test_browser_private_navigation.py

```diff
@@ -16,12 +17,20 @@ def setUp(self):
         self.test_url = self.marionette.absolute_url('mozilla.html')
 
     def test_browser_private_navigation(self):
-        search = Search(self.marionette)
-        search.launch()
-        private_window = search.open_new_private_window()
+        self.device.hold_home_button()
+
+        cards_view = CardsView(self.marionette)
+        private_window = cards_view.open_new_private_window()
+
         browser = private_window.go_to_url(self.test_url)
         browser.switch_to_content()
         Wait(self.marionette).until(lambda m: m.title == 'Mozilla')
 
+        self.device.hold_home_button()
+
+        cards_view = CardsView(self.marionette)
+        cards_view.wait_for_cards_view()
+        search = cards_view.open_new_browser()
+        # This is to switch the marionette context to the browser content
```

Comment:
This looks odd. I wonder if we counldn't put the frame switch in `open_new_browser()`. What do you think?